apple/internal/aspects/resource_aspect.bzl

```diff
@@ -120,6 +125,12 @@ def _apple_resource_aspect_impl(target, ctx):
         collect_args["res_attrs"] = ["data"]
         owner = str(ctx.label)
 
+    elif ctx.rule.kind in ["apple_static_framework_import", "apple_static_xcframework_import"]:
```

Comment:
Does this need a `kind` check, versus just looking for the provider? Also, it seems odd that we need a new provider versus using the existing `AppleResourceInfo` infrastructure.

Reply:
Yes, because the provider is only present when there are bundle resource files inside the .framework.  The aspect also needs to collect resources from the `data` attribute, similar to `objc_library/swift_library`:

```starlark
collect_args["res_attrs"] = ["data"]
owner = str(ctx.label)
```

In order to use `AppleResourceInfo`, we would need to collect and merge resources from `data` inside both `apple_static_framework_import` and `apple_static_xcframework_import` rules. I chose to delegate this logic to `apple_resource_aspect`, as the other rules do.

Reply:
Do we need the kind check though if we are keeping the provider? Couldn't moving this outside the kind checks, and looking for the provider, allow custom rules to add support for this as well?

Reply:
Nevermind, I see.